### PR TITLE
fix(wmg): don't send requests if only genes or tissues have been removed

### PR DESCRIPTION
**Pull request by seve**
- #3800 
### Reviewers
**Functional:**  @tihuan

### CODE REVIEW

File: frontend/src/views/WheresMyGene/components/HeatMap/components/YAxisChart/index.tsx [hunk 1]
```diff
@@ -216,9 +214,12 @@ export default memo(function YAxisChart({
       {yAxisInfoCoords &&
         isMarkerGenes &&
         !FMG_EXCLUDE_TISSUES.includes(tissue) &&
+        cellTypeMetadata.length > 0 &&
```
Comment (seve) [inline frontend/src/views/WheresMyGene/components/HeatMap/components/YAxisChart/index.tsx:217]: fixes #3849?

Comment (tihuan) [reply frontend/src/views/WheresMyGene/components/HeatMap/components/YAxisChart/index.tsx:217]: This quick fix stops the runtime error, but  do we know why `cellTypeMetadata` can be an empty array or undefined here? In my mind every cell type should always have serialized metadata, so `cellTypeMetadata` should always be available 🤔

**Review by tihuan — APPROVED** [review]
Thanks for the great solution, Seve! LGTM 🚀

File: frontend/src/views/WheresMyGene/components/HeatMap/components/YAxisChart/index.tsx (hunk 1 above)
Comment (seve) [reply frontend/src/views/WheresMyGene/components/HeatMap/components/YAxisChart/index.tsx:217]: Just for posterity I followed the thread earlier when I encountered the bug and saw that `cellTypeMetadata` is empty when `cellTypes` is empty

Comment (tihuan) [reply frontend/src/views/WheresMyGene/components/HeatMap/components/YAxisChart/index.tsx:217]: Thanks for digging into this! 

Interesting, the surrounding code was introduced when the FMG info button was added, how come we're only seeing the bug now? And does the runtime error only happen for certain datasets or all of them? 

IIRC the info buttons were rendering just fine before the holidays, so not sure what code change caused the runtime error now 🤔

Comment (tihuan) [reply frontend/src/views/WheresMyGene/components/HeatMap/components/YAxisChart/index.tsx:217]: CC @atarashansky in case you have more info here! Thanks!

Comment (seve) [reply frontend/src/views/WheresMyGene/components/HeatMap/components/YAxisChart/index.tsx]: fix is void from upstream changes on main

**Pull request merged**